RELEASE/scripts/autoscend/auto_choice_adv.ash

```diff
@@ -90,6 +171,175 @@ void main(int choice, string page)
+				run_choice(3); // relocate lawyers to park
+			} else if (have_effect($effect[Thrice-Cursed]) > 0) {
+				run_choice(1); // fight the spirit
+			} else {
+				run_choice(2); // get cursed
+			}
+			break;
+		case 781: // Earthbound and Down (An Overgrown Shrine (Northwest))
+			if (get_property("hiddenApartmentProgress").to_int() == 0) {
+				run_choice(1); // unlock the Hidden Apartment Building
+			} else if (item_amount($item[moss-covered stone sphere]) > 0) {
+				run_choice(2); // get the stone triangle
+			} else {
+				run_choice(6); // skip
+			}
+			break;
+		case 783: // Water You Dune (An Overgrown Shrine (Southwest))
+			if (get_property("hiddenHospitalProgress").to_int() == 0) {
+				run_choice(1); // unlock the Hidden Hospital
+			} else if (item_amount($item[dripping stone sphere]) > 0) {
+				run_choice(2); // get the stone triangle
+			} else {
+				run_choice(6); // skip
+			}
+			break;
+		case 784: // You, M. D. (The Hidden Hospital)
+			run_choice(1); // fight the spirit
+			break;
+		case 785: // Air Apparent (An Overgrown Shrine (Northeast))
+			if (get_property("hiddenOfficeProgress").to_int() == 0) {
+				run_choice(1); // unlock the Hidden Office Building
+			} else if (item_amount($item[crackling stone sphere]) > 0) {
+				run_choice(2); // get the stone triangle
+			} else {
+				run_choice(6); // skip
+			}
+			break;
+		case 786: // Working Holiday (The Hidden Office Building)
+			if (item_amount($item[McClusky File (Complete)]) > 0) {
+				run_choice(1); // fight the spirit
+			} else if (item_amount($item[Boring Binder Clip]) == 0) {
+				run_choice(2); // get boring binder clip
+			} else {
+				run_choice(3); // fight an accountant
+			}
+			break;
+		case 787: // Fire When Ready (An Overgrown Shrine (Southeast))
+			if (get_property("hiddenBowlingAlleyProgress").to_int() == 0) {
+				run_choice(1); // unlock the Hidden Bowling Alley
+			} else if (item_amount($item[scorched stone sphere]) > 0) {
+				run_choice(2); // get the stone triangle
+			} else {
+				run_choice(6); // skip
+			}
+			break;
+		case 788: // Life is Like a Cherry of Bowls (The Hidden Bowling Alley)
+			run_choice(1); // bowl for stats 4 times then fight the spirit on 5th occurrence
+			break;
+		case 789: // Where Does The Lone Ranger Take His Garbagester? (The Hidden Park)
+			if (get_property("relocatePygmyJanitor").to_int() != my_ascensions()) {
+				run_choice(2); // Relocate the Pygmy Janitor to the park
+			} else {
+				run_choice(1); // Get Hidden City zone items
+			}
+			break;
+		case 791: // Legend of the Temple in the Hidden City (A Massive Ziggurat)
+			if (item_amount($item[stone triangle]) == 4) {
+				run_choice(1); // fight the Protector Spirit (or replacement)
+			} else {
+				run_choice(2); // skip
+			}
+			break;
+		case 875: // Welcome To Our ool Table (The Haunted Billiards Room)
+			if (currentPoolSkill() >= 18) {

```

Comment:
Need to add something like the following (i am using pseudocode)
```
int poolSkillPracticeGains()
{
//counts the predicted gains from practicing your pool skill on choice adventure 875
 int count = 1;
 if(has effect chalky hand) count += 1;
 if(currently equipped staff of fats) count += 2;
 return count;
}
```

Then for this line it should be.
`if(poolSkillPracticeGains() == 1 || currentPoolSkill() > 16)`
That is just the times where it is always better to do choice 1.
If we want statistically better than that 16 could be even lower (although then you need to calculate the exact statistical figures... which I might do later. for now I think it can be guesstimated at 15)

The reason is that if you do choice 1. on success you get key, on failure you get +1 skill.
Chances of success are:
yourpoolskill = success%
13 = 0%
14 = 20%
15 = 40%
16 = 60%
17 = 80%
18 = 100%

So if your pool skill is 17, there is no point in ever doing choice 2. because choice 1 has 80% chance of success, on a failure the next attempt will be 100% success.
while taking choice 2 means 0% chance of finishing this room this turn, and 100% next time you get the NC.
Thus waiting until 18 is too late.

If your practice gains are just 1. Then there is no reason to use choice 2, your gains are the same for both options. while choice 1 gives you a progressive odd of early completion starting from 14 pool skill. (20%, then 40%, then 60%, etc)

Reply:
Nice, I'll look into this. Thanks for the feedback. I'd just moved stuff without too much thought into how to make this better but this is a good improvement.

Reply:
This can also go in a followup PR if you prefer. I can go ahead and make it after you merge this

Reply:
created said subsequent PR #383
ended up having to revamp the code for adventuring in that area too rather than only handling the NC